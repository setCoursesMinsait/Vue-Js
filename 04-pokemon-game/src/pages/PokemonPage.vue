<template>
    <h1 v-if="!pokemon">Espere por favor ...</h1>
    <div v-else>
        <h1>¿Quién es este pokémon?</h1>
        <!-- TODO: img -->
        <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon"/>
        <!-- TODO: opciones -->
        <PokemonOptions :pokemons="pokemonArr"
            @selection="checkAnswer"/>
        
        <template v-if="showAnswer">
            <h2 class="fade-in"> {{message}} </h2>
            <button @click="newGame">Nuevo Juego</button>
        </template>
    </div>
</template>

<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'

import getPokemonOptions from '@/helpers/getPokemonOptions'

export default {
    components: {PokemonPicture, PokemonOptions},
    data() {
        return {
            pokemonArr: [],
            pokemon: null,
            showPokemon: false,
            showAnswer: false,
            message: ''
        }
    },
    methods: {
        async mixPokemonArray() {
            this.pokemonArr = await getPokemonOptions()

            const randomInt = Math.floor( Math.random() * 4 )
            this.pokemon = this.pokemonArr[ randomInt ]
        },
        checkAnswer( pokemonId ) {
            this.showPokemon = true
            this.showAnswer = true
            if( pokemonId === this.pokemon.id){
                this.message = `Correcto, ${this.pokemon.name}`
            } else {
                this.message = `Oopss, era ${this.pokemon.name}`
            }
        },
        newGame() {
            this.showPokemon = false
            this.showAnswer = false
            this.pokemonArr = []
            this.pokemon = null
            this.mixPokemonArray()
        }
    },
    mounted() {
        this.mixPokemonArray()
    }
}
</script>