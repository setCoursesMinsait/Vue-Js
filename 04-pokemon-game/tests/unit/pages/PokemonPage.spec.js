import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage";

describe( 'PokemonPage Component', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallowMount( PokemonPage )
  })

  test( 'debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test( 'debe de llamar mixPokemonArray al montar', () => {
    const mixPokemonArraySpy = jest.spyOn( PokemonPage.methods, 'mixPokemonArray')
    shallowMount(PokemonPage)

    expect(mixPokemonArraySpy).toHaveBeenCalled()
  })
})