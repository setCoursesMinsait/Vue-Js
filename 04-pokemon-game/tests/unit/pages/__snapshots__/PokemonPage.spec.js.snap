// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe de hacer match con el snapshot 1`] = `<h1>Espere por favor ...</h1>`;
